src/frontend/src/helpers/Compiler/Interface.ts

```diff
@@ -1,38 +1,69 @@
-import {ProjectID, FileID, File} from "../Storage/Interface";
+import {AbstractStorage,
+        ProjectID,
+        FileID,
+        FileBrief,
+        File,
+        ext,
+        basename} from "../Storage/Interface";
+import {groupBy} from "../utils";
 
 export {AbstractCompiler,
         CompilerResult,
         CompilerMessage,
-        Test,
-        PID};
+        TestBrief,
+        Test};
 
 abstract class AbstractCompiler {
```

Comment:
Probably want a is_running() method so that the UI can tell if something's running.